runtime/zeitgeist/src/lib.rs

```diff
@@ -142,6 +146,16 @@ impl Contains<RuntimeCall> for IsCallable {
                     _ => true,
                 }
             }
+            // Permissioned contracts: Only deployable via utility.dispatch_as(...)
+            RuntimeCall::Contracts(inner_call) => match inner_call {
+                call { .. } => true,
+                call_old_weight { .. } => true,
+                instantiate { .. } => true,
+                instantiate_old_weight { .. } => true,
+                remove_code { .. } => true,
+                set_code_contracts { .. } => true,
+                _ => false,
+            },
```

Comment:
Ok, I scanned all possible calls from [here](https://github.com/paritytech/substrate/blob/5ea6d95309aaccfa399c5f72e5a14a4b7c6c4ca1/frame/contracts/src/lib.rs#L365-L676). Seems like your call filter does apply on every extrinsic which allows to upload a `code: Vec<u8>`, but the utility pallet can handle those dispatches. I see 👍

Reply:
`dispatch_bypass_filter` is only allowed for the root origin ✅